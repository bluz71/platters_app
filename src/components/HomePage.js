import React, { Component } from 'react';
import axios from 'axios';
import { API_HOST } from '../config';

const HOME_ENDPOINT = `${API_HOST}/home.json`;

class HomePage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      albumOfTheDay: {},
      mostRecentAlbums: [],
      mostRecentComments: []
    };
  }

  componentDidMount() {
    this.getHome();
  }

  getHome() {
    axios.get(HOME_ENDPOINT)
      .then(response => {
        this.setState({
          albumOfTheDay: response.data.album_of_the_day,
          mostRecentAlbums: response.data.most_recent.albums,
          mostRecentComments: response.data.most_recent.comments
        });
      });
  }

  render() {
    return (
      <h1>Home Page</h1>
    );
  }
}

export default HomePage;
