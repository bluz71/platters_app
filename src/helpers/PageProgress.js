import NProgress from 'nprogress'

class PageProgress {
  constructor () {
    this.loaded = false
  }

  start () {
    if (!this.loaded) {
      NProgress.start()
    }
  }

  done () {
    if (!this.loaded) {
      this.loaded = true
      NProgress.done()
    }

    return true
  }
}

export default PageProgress
