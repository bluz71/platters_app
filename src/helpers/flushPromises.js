// Details:
//   https://blog.rescale.com/testing-promise-side-effects-with-asyncawait
//   https://github.com/facebook/jest/issues/2157#issuecomment-279171856

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

export default flushPromises;
