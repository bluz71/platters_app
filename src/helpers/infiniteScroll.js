// Code snippet based on:
//   http://blog.sodhanalibrary.com/2016/08/detect-when-user-scrolls-to-bottom-of.html
//
const infiniteScoll = (pagination, onPageEnd) => {
  const moreToScroll = pagination && pagination.next_page;
  if (!moreToScroll) {
    return;
  }

  const windowHeight = window.innerHeight;
  const body = document.body;
  const html = document.documentElement;
  const docHeight =
    Math.max(
      body.scrollHeight,
      body.offsetHeight,
      html.clientHeight,
      html.scrollHeight,
      html.offsetHeight
    ) - 20;
  const windowBottom = windowHeight + window.pageYOffset;

  if (windowBottom >= docHeight) {
    onPageEnd();
  }
};

export default infiniteScoll;
