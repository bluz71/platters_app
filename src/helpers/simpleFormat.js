// Code snippet provided by:
//   https://makandracards.com/makandra/1395-simple_format-helper-for-javascript
//
const simpleFormat = str => {
  str = str.replace(/\r\n?/, '\n')
  str = str.trim()

  if (str.length > 0) {
    str = str.replace(/\n\n+/g, '</p><p>')
    str = str.replace(/\n/g, '<br />')
    str = `<p>${str}</p>`
  }

  return str
}

export default simpleFormat
