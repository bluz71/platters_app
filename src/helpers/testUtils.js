import { appAuth } from '../lib/appAuth';

// Details:
//   https://blog.rescale.com/testing-promise-side-effects-with-asyncawait
//   https://github.com/facebook/jest/issues/2157#issuecomment-279171856
const flushPromises = () => new Promise(resolve => setImmediate(resolve));

const logInUser = () => (
  appAuth.logIn('eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjo1MywibmFtZSI6ImRlbm5pc2I1NSI\
sImFkbWluIjpmYWxzZSwiZXhwIjoxNTM3ODQ4NDA0LCJpc3MiOiJwbGF0dGVycyIsImF1ZCI6InBs\
YXR0ZXJzX2FwcCJ9.ZUV69wWVXfiiMtQ2ZFPvgqvhZWwQ4WC1bI15Q9DpFCU')
);

const logOutUser = () => (
  appAuth.logOut()
);

export { flushPromises, logInUser, logOutUser };
