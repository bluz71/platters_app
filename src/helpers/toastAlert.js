import { toast } from 'react-toastify';

const toastAlert = (message) => {
  toast.error(message, { className: 'ToastAlert' });
};

export default toastAlert;
