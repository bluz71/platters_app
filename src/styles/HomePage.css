.Home .first {
  margin-top: 1em;
}

.Home h1 {
  text-align: center;
}

.Home .logo img {
  height: 24em;
}

.Home .dark {
  background-color: #333;
  border-radius: 6px;
}

.Home .dark h2 {
  text-align: center;
  color: #eee;
  margin: 1.5em;
}

.Home .dark span {
  font-size: 3em;
  margin-bottom: 0.5em;
  color: #eee;
}

.Home .dark p {
  font-size: 1.4em;
  font-weight: lighter;
  color: #eee;
  margin-bottom: 1.5em;
}

.Home .dark .complete-details {
  margin-bottom: 1.5em;
}

.Home .album-of-the-day .page-header {
  margin-top: 0.75em;
}

.Home .album-of-the-day a {
  color: #111;
  text-decoration: none;
}

.Home .album-of-the-day a:hover {
  color: #3c9a5f;
}

.Home .album-of-the-day img {
  max-height: 400px;
  max-width: 400px;
  height: 100%;
  width: 100%;
  margin-top: 1em;
  margin-bottom: 1em;
  border: 1px solid #777;
  border-radius: 4px;
}

.Home .album-of-the-day img:hover {
  border: 1px solid #3c9a5f;
}
