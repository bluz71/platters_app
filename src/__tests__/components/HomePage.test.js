import React from 'react';
import { shallow } from 'enzyme';
import { flushPromises, logInUser } from '../../helpers/testUtils';
import HomePage from '../../components/HomePage.js';

describe('<HomePage />', () => {
  it('renders without crashing', () => {
    shallow(<HomePage />);
  });

  it('renders home page', async () => {
    const wrapper = shallow(<HomePage />);
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders home page when logged in', async () => {
    logInUser();
    const wrapper = shallow(<HomePage />);
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
