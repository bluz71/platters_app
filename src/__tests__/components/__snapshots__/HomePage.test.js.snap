// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders home page 1`] = `
<div
  className="Home"
>
  <div>
    <Row
      bsClass="row"
      className="first"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={7}
      >
        <Jumbotron
          bsClass="jumbotron"
          componentClass="div"
        >
          <h1>
            Platters
          </h1>
          <p>
            An album collection web application developed using modern web technologies.
          </p>
          <div
            className="text-center"
          >
            <Link
              className="btn btn-success btn-lg"
              replace={false}
              to="/about"
            >
              About
            </Link>
            <Link
              className="btn btn-success btn-lg spacer-left-tiny"
              replace={false}
              to="/sign_up"
            >
              Sign Up
            </Link>
          </div>
        </Jumbotron>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={5}
      >
        <div
          className="logo"
        >
          <img
            alt="Platters"
            className="img-responsive center-block"
            src="platters-black.svg"
          />
        </div>
      </Col>
    </Row>
    <Row
      bsClass="row"
      className="dark"
      componentClass="div"
    >
      <h2>
        Prime technologies used by the Platters application.
      </h2>
      <Col
        bsClass="col"
        className="text-center"
        componentClass="div"
        md={4}
      >
        <FontAwesome
          name="code"
        />
        <p>
          React JavaScript front-end user interface library
        </p>
      </Col>
      <Col
        bsClass="col"
        className="text-center"
        componentClass="div"
        md={4}
      >
        <FontAwesome
          name="diamond"
        />
        <p>
          Ruby on Rails back-end API server
        </p>
      </Col>
      <Col
        bsClass="col"
        className="text-center"
        componentClass="div"
        md={4}
      >
        <FontAwesome
          name="tablet"
        />
        <p>
          Bootstrap responsive front-end web framework.
        </p>
      </Col>
      <div
        className="complete-details text-center"
      >
        <Link
          className="btn btn-success"
          replace={false}
          to="/details"
        >
          Complete Details
        </Link>
      </div>
    </Row>
  </div>
  <div
    className="album-of-the-day"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={12}
      >
        <h2
          className="page-header"
        >
          Album of the day - 
          <Link
            replace={false}
            to="/artist/xyz/album/album_of_the_day"
          >
            Album of the Day
          </Link>
           
          <small>
            by 
            <Link
              replace={false}
              to="/artist/xyz"
            >
              XYZ
            </Link>
          </small>
        </h2>
        <Jumbotron
          bsClass="jumbotron"
          componentClass="div"
        >
          <img
            alt="Album of the Day"
            className="img-responsive center-block"
            src="http://localhost:3000/uploads/album/cover/832/XYZ--Album_of_the_Day.jpg"
          />
        </Jumbotron>
      </Col>
    </Row>
  </div>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={12}
    >
      <h2
        className="page-header"
      >
        New Albums
      </h2>
      <AlbumsList
        albums={
          Array [
            Object {
              "album_slug": "abc",
              "artist": "ABC-Artist",
              "artist_slug": "abc_artist",
              "comments_count": 41,
              "cover_url": "http://localhost:3000/uploads/album/cover/1/small_abc--abc.jpg",
              "genre": "Rock",
              "id": 1,
              "title": "ABB",
              "tracks": Array [
                "1. First",
                "2. Second",
                "3. Third",
                "4. Fourth",
                "5. Fifth",
                "6. Sixth",
                "...",
              ],
              "tracks_count": 11,
              "year": 2008,
            },
            Object {
              "album_slug": "def",
              "artist": "DEF-Artist",
              "artist_slug": "def_artist",
              "comments_count": 44,
              "cover_url": "http://localhost:3000/uploads/album/cover/978/small_def--def.jpg",
              "genre": "Classical",
              "id": 2,
              "title": "DEF",
              "tracks": Array [
                "1. One",
                "2. Two",
              ],
              "tracks_count": 27,
              "year": 2015,
            },
          ]
        }
        onGenre={[Function]}
        onYear={[Function]}
      />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={12}
    >
      <h2
        className="page-header"
      >
        New Comments
      </h2>
      <CommentsList
        comments={
          Map {
            20 => Object {
              "artist": "ABC-Artist",
              "body": "A comment",
              "created_at": "<time datetime=\\"2018-04-22T05:50:26Z\\" data-local=\\"time-ago\\">April 22, 2018  5:50am</time>",
              "for": "Artist",
              "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
              "id": 20,
              "name": "ABC-Artist",
              "path": "artist/abc_artist",
              "user_name": "user1",
              "user_slug": "user1",
            },
            21 => Object {
              "artist": "ABC-Artist",
              "body": "Another comment",
              "created_at": "<time datetime=\\"2018-04-22T05:12:25Z\\" data-local=\\"time-ago\\">April 22, 2018  5:12am</time>",
              "for": "Artist",
              "gravatar_url": "https://gravatar.com/avatar/699b7e1dd13a6638d847f2d7acf94843?s=80&r=pg&d=identicon",
              "id": 21,
              "name": "ABC-Artist",
              "path": "artist/abc_artist",
              "user_name": "admin",
              "user_slug": "admin",
            },
            22 => Object {
              "artist": "ABC",
              "body": "A final comment",
              "created_at": "<time datetime=\\"2018-04-14T04:29:39Z\\" data-local=\\"time-ago\\">April 14, 2018  4:29am</time>",
              "for": "Album",
              "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
              "id": 22,
              "name": "ABC",
              "path": "artist/abc_artist/album/abc",
              "user_name": "user1",
              "user_slug": "user1",
            },
          }
        }
        onDeleteComment={[Function]}
      />
    </Col>
  </Row>
</div>
`;

exports[`<HomePage /> renders home page when logged in 1`] = `
<div
  className="Home"
>
  <div
    className="album-of-the-day"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={12}
      >
        <h2
          className="page-header"
        >
          Album of the day - 
          <Link
            replace={false}
            to="/artist/xyz/album/album_of_the_day"
          >
            Album of the Day
          </Link>
           
          <small>
            by 
            <Link
              replace={false}
              to="/artist/xyz"
            >
              XYZ
            </Link>
          </small>
        </h2>
        <Jumbotron
          bsClass="jumbotron"
          componentClass="div"
        >
          <img
            alt="Album of the Day"
            className="img-responsive center-block"
            src="http://localhost:3000/uploads/album/cover/832/XYZ--Album_of_the_Day.jpg"
          />
        </Jumbotron>
      </Col>
    </Row>
  </div>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={12}
    >
      <h2
        className="page-header"
      >
        New Albums
      </h2>
      <AlbumsList
        albums={
          Array [
            Object {
              "album_slug": "abc",
              "artist": "ABC-Artist",
              "artist_slug": "abc_artist",
              "comments_count": 41,
              "cover_url": "http://localhost:3000/uploads/album/cover/1/small_abc--abc.jpg",
              "genre": "Rock",
              "id": 1,
              "title": "ABB",
              "tracks": Array [
                "1. First",
                "2. Second",
                "3. Third",
                "4. Fourth",
                "5. Fifth",
                "6. Sixth",
                "...",
              ],
              "tracks_count": 11,
              "year": 2008,
            },
            Object {
              "album_slug": "def",
              "artist": "DEF-Artist",
              "artist_slug": "def_artist",
              "comments_count": 44,
              "cover_url": "http://localhost:3000/uploads/album/cover/978/small_def--def.jpg",
              "genre": "Classical",
              "id": 2,
              "title": "DEF",
              "tracks": Array [
                "1. One",
                "2. Two",
              ],
              "tracks_count": 27,
              "year": 2015,
            },
          ]
        }
        onGenre={[Function]}
        onYear={[Function]}
      />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={12}
    >
      <h2
        className="page-header"
      >
        New Comments
      </h2>
      <CommentsList
        comments={
          Map {
            20 => Object {
              "artist": "ABC-Artist",
              "body": "A comment",
              "created_at": "<time datetime=\\"2018-04-22T05:50:26Z\\" data-local=\\"time-ago\\">April 22, 2018  5:50am</time>",
              "for": "Artist",
              "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
              "id": 20,
              "name": "ABC-Artist",
              "path": "artist/abc_artist",
              "user_name": "user1",
              "user_slug": "user1",
            },
            21 => Object {
              "artist": "ABC-Artist",
              "body": "Another comment",
              "created_at": "<time datetime=\\"2018-04-22T05:12:25Z\\" data-local=\\"time-ago\\">April 22, 2018  5:12am</time>",
              "for": "Artist",
              "gravatar_url": "https://gravatar.com/avatar/699b7e1dd13a6638d847f2d7acf94843?s=80&r=pg&d=identicon",
              "id": 21,
              "name": "ABC-Artist",
              "path": "artist/abc_artist",
              "user_name": "admin",
              "user_slug": "admin",
            },
            22 => Object {
              "artist": "ABC",
              "body": "A final comment",
              "created_at": "<time datetime=\\"2018-04-14T04:29:39Z\\" data-local=\\"time-ago\\">April 14, 2018  4:29am</time>",
              "for": "Album",
              "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
              "id": 22,
              "name": "ABC",
              "path": "artist/abc_artist/album/abc",
              "user_name": "user1",
              "user_slug": "user1",
            },
          }
        }
        onDeleteComment={[Function]}
      />
    </Col>
  </Row>
</div>
`;
